modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1322,15 +1338,11 @@ private com.liferay.object.model.ObjectEntry _upsertNestedObjectEntry(
 		long groupId = getGroupId(
 			relatedObjectDefinition, relatedObjectDefinition.getScope());
 
-		ObjectEntry objectEntry = new ObjectEntry();
-
-		objectEntry.setProperties(nestedObjectEntryProperties);
-
-		String externalReferenceCode = (String)nestedObjectEntryProperties.get(
-			"externalReferenceCode");
+		ObjectEntry objectEntry = _parseToDTOObjectEntry(
```

Comment:
Careful, If I understood correctly, the idea of having this parser was also to not drag the `nestedObjectEntryProperties` map through the methods but create directly the ObjectEntry so we can manage an Entity that we know because it is defined by business logic.

cc/ @ccorreagg

Reply:
Make sense @LuismiBarcos! I didn't take this into account

Reply:
yes! exacty, I meant creating the ObjectEntry not so deeper in the code, but before, to avoid managing Maps.

Let me try some commits to see if it makes more sense